featuretools/tests/feature_discovery/test_feature_discovery.py

```diff
@@ -0,0 +1,384 @@
+                "Double,numeric": ["f3"],
+                "Boolean": ["f4"],
+                "BooleanNullable": ["f5"],
+            },
+            Lag,
+            [["f2", "f1"], ["f3", "f1"], ["f4", "f1"], ["f5", "f1"]],
+        ),
+    ],
+)
+def test_get_matching_features(feature_groups, primitive, expected):
+    actual = get_matching_features(feature_groups, primitive)
+
+    assert actual == expected
+
+
+@pytest.mark.parametrize(
+    "col_defs, primitives, expected",
+    [
+        (
+            [
+                ("f_1", "Double"),
+                ("f_2", "Double"),
+                ("f_3", "Boolean"),
+                ("f_4", "Double"),
+            ],
+            [AddNumeric],
+            {"f_1 + f_2", "f_1 + f_4", "f_2 + f_4"},
+        ),
+        (
+            [
+                ("f_1", "Double"),
+                ("f_2", "Double"),
+            ],
+            [Absolute],
+            {"ABSOLUTE(f_1)", "ABSOLUTE(f_2)"},
+        ),
+    ],
+)
+def test_new_dfs(col_defs, primitives, expected):
+    input_feature_names = set([x[0] for x in col_defs])
+    df = generate_fake_dataframe(
+        col_defs=col_defs,
+    )
+
+    all_features = my_dfs(df.ww.schema, primitives)
+
+    new_feature_names = set([x.name for x in all_features]) - input_feature_names
+    assert new_feature_names == expected
+
+
+def get_default_logical_type(tags: Set[str]):
+    for tag, lt in DEFAULT_LT_FOR_TAG.items():
+        if tag in tags:
+            return lt
+
+    raise Exception(f"NO DEFAULT LOGICAL TYPE FOR TAGS: {tags}")
+
+
+@pytest.mark.parametrize(
+    "col_defs, primitives",
+    [
+        (
+            [
+                ("idx", "Double", {"index"}),
+                ("f_1", "Double"),
+                ("f_2", "Double"),
+                ("f_3", "Boolean"),
+                ("f_4", "Boolean"),
+                ("f_5", "Double"),
+            ],
+            [AddNumeric, Absolute, SubtractNumeric],
+        ),
+        (
+            [
+                ("idx", "Double", {"index"}),
+                ("t_idx", "Datetime", {"time_index"}),
+                ("f_2", "Double"),
+                ("f_3", "Boolean"),
+                ("f_4", "Boolean"),
+                ("f_5", "Double"),
+            ],
+            [Lag, Day, Absolute, AddNumeric],
+        ),
+    ],
+)
+def test_compare_dfs(col_defs, primitives):
+    input_feature_names = set([x[0] for x in col_defs])
+    df = generate_fake_dataframe(
+        col_defs=col_defs,
+    )
+
+    es = EntitySet(id="nums")
+    es.add_dataframe(df, "nums", index="idx")
+
+    features_old = dfs(
+        entityset=es,
+        target_dataframe_name="nums",
+        trans_primitives=primitives,
+        features_only=True,
+    )
+
+    features_new = my_dfs(df.ww.schema, primitives)
+
+    feature_names_old = set([x.get_name() for x in features_old]) - input_feature_names
+
+    feature_names_new = set([x.name for x in features_new]) - input_feature_names
```

Comment:
```suggestion
    feature_names_old = set([x.get_name() for x in features_old]) - input_feature_names

    feature_names_new = set([x.name for x in features_new]) - input_feature_names
```
Do we need to subtract from both if we are comparing them for equality right after?